code/dijkstra-algorithm/ViewModel/src/main/java/ru/unn/agile/dijkstraalgorithm/viewmodel/ViewModel.java

```diff
@@ -106,56 +133,129 @@ private void updateVertexList() {
     public void calculatePath() {
         String toPath = getVertexTo();
         String fromPath = getVertexFrom();
+        StringBuilder messageCalculate = new StringBuilder(LogMessages.CALCULATE_WAS_PRESSED);
         if (toPath == null || fromPath == null) {
+            messageCalculate.append("unsuccessfully. Check From and To vertexes.");
+            logger.log(messageCalculate.toString());
+            updateLogs();
             return;
         }
 
         graph.calculate(fromPath);
+        messageCalculate.append("successfully.");
+        logger.log(messageCalculate.toString());
+        updateLogs();
+
         resultPath.setValue(graph.getPath(toPath));
     }
 
     public StringProperty vertex1Property() {
         return vertex1;
     }
-
     public StringProperty vertex2Property() {
         return vertex2;
     }
-
     public StringProperty weightProperty() {
         return weight;
     }
-
     private String getVertexFrom() {
         return vertexFrom.get();
     }
-
     public StringProperty vertexFromProperty() {
         return vertexFrom;
     }
-
-
     private String getVertexTo() {
         return vertexTo.get();
     }
-
     public StringProperty vertexToProperty() {
         return vertexTo;
     }
-
     public StringProperty resultPathProperty() {
         return resultPath;
     }
-
     public SimpleBooleanProperty addingNewEdgeDisabledProperty() {
         return addingNewEdgeDisabled;
     }
-
     public ObservableList<EdgeViewModel> getEdgeList() {
         return edgeList;
     }
-
     public ObservableList<String> getVertexList() {
         return vertexList;
     }
+    public StringProperty logsProperty() {
+        return logs;
+    }
+    public final String getLogs() {
+        return logs.get();
+    }
+    public final List<String> getLogList() {
+        return logger.getLog();
```

Comment:
If your class are initialized via empty constructor, you will get NullPointerException by calling this method.
So it's safer to check `logger` for null first, and then call `getLog` method

Reply:
Done